test/e2e/thv-operator/virtualmcp/virtualmcp_auth_discovery_test.go

```diff
@@ -0,0 +1,615 @@
+package virtualmcp
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"time"
+
+	"github.com/mark3labs/mcp-go/client"
+	"github.com/mark3labs/mcp-go/mcp"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	mcpv1alpha1 "github.com/stacklok/toolhive/cmd/thv-operator/api/v1alpha1"
+)
+
+var _ = Describe("VirtualMCPServer Auth Discovery", Ordered, func() {
+	var (
+		testNamespace   = "default"
+		mcpGroupName    = "test-auth-discovery-group"
+		vmcpServerName  = "test-vmcp-auth-discovery"
+		backend1Name    = "backend-with-token-exchange"
+		backend2Name    = "backend-with-header-injection"
+		backend3Name    = "backend-no-auth"
+		authConfig1Name = "test-token-exchange-auth"
+		authConfig2Name = "test-header-injection-auth"
+		authSecret1Name = "test-token-exchange-secret"
+		authSecret2Name = "test-header-injection-secret"
+		timeout         = 5 * time.Minute
+		pollingInterval = 5 * time.Second
+		mockServer      *httptest.Server
+	)
+
+	BeforeAll(func() {
+		By("Setting up mock HTTP server for fetch tool testing")
+		mockServer = httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
+			w.WriteHeader(http.StatusOK)
```

Comment:
Do we expect that this server will receive auth tokens? If yes, could we assert that it does?